### PR TITLE
Validator Tracing and TimeTriggeredValidator

### PR DESCRIPTION
**Pull request by mikand**
In this PR we extend the internal sequential validation engine to return the sequence of states encountered by executing a sequential plan and we add support for validation (and tracing) of TimeTriggeredPlans

### CODE REVIEW

**Comment by karpase** [comment]
Commenting just to say I would like to see this implemented - I have some experiments this could really help with.

**Comment by Framba-Luca** [comment]
In pr #509 also happens a "transformation" of a temporal structure into a non-temporal one. After the merge of both there will most certainly be space for refactoring (that maybe can be considered also when designing the missing code).

**Review by Framba-Luca — CHANGES_REQUESTED** [review]
LGTM! :smiley:

I left 4 comments. 2 are minor code-level improvements, 1 I think it's a bug, the 4th one is just a note on the `SequentialValidator` `supported_kind`

**Review by Framba-Luca — APPROVED** [review]

**Review by arbimo — APPROVED** [review]
LGTM! I just added a small comment about an awkwardness in error handling.

File: unified_planning/engines/plan_validator.py [hunk 1]
```diff
@@ -187,42 +155,47 @@ def _validate(
             except UPInvalidActionError as e:
                 msg = f"{str(i)}-th action instance {str(ai)} creates an Invalid Action: {str(e)}"
             try:
```
[the remark this answers is not in the document]
Comment (alvalentini) [reply unified_planning/engines/plan_validator.py]: You are right! Fixed it!

[the remark this answers is not in the document]
Comment (arbimo) [reply unified_planning/engines/plan_validator.py]: Thanks!  I'll merge it then ;)

**Pull request merged**